Rubberduck.Parsing/UIContext/UiDispatcher.cs

```diff
@@ -53,16 +41,40 @@ public void Invoke(Action action)
             }
         }
 
+        /// <inheritdoc />
+        public void FlushMessageQueue()
+        {
+            CheckInitialization();
+
+            void PumpMessages()
+            {
+                var message = new NativeMethods.NativeMessage();
+                var handle = GCHandle.Alloc(message);
+
+                while (NativeMethods.PeekMessage(ref message, IntPtr.Zero, 0, 0, NativeMethods.PeekMessageRemoval.Remove))
+                {
+                    NativeMethods.TranslateMessage(ref message);
+                    NativeMethods.DispatchMessage(ref message);
+                }
+
+                handle.Free();
+            }
+
+            if (_contextProvider.UiContext == SynchronizationContext.Current)
+            {
+                PumpMessages();
```

Comment:
FWIW, I tripped over this as at first read, I thought I was reading a recursive method but now I see that it's a local function. I assume you elected for a local so it wouldn't be used anywhere else but this method. Maybe consider moving it to be the very first or the very last thing within the method and provide a comment?

Reply:
You hit the nail on the head re why it's a local function. I'll move it to the bottom (and comment) - I think you're right about it looking "less like recursion" there.

Reply:
At least, with the new build setup, they seem to auto-generated the appropriate code again.